shlex.js

```diff
@@ -1,45 +1,14 @@
 'use strict'
 
 /*
-Port of a subset of the features of CPython's shlex module, which provides a
-shell-like lexer. Original code by Eric S. Raymond and other contributors.
+  Port of a subset of the features of CPython's shlex module, which provides a
+  shell-like lexer. Original code by Eric S. Raymond and other contributors.
 */
 
-var assert = require('assert')
-
-
 class Shlexer {
-  constructor (stream, posix = true, punctuation_chars = '') {
+  constructor (string) {
     this.i = 0
-    this.stream = stream
-
-    this.posix = posix
-    if (this.posix !== true) {
-      assert.fail('Only POSIX mode is implemented')
-    }
-
-    /**
-     * The string of characters that are recognized as comment beginners. All
-     * characters from the comment beginner to end of line are ignored.
-     */
-    this.commenters = ''
-    if (this.commenters !== '') {
-      assert.fail('Comments are not implemented yet')
-    }
-
-    /**
-     * The string of characters that will accumulate into multi-character
-     * tokens. By default, includes all ASCII alphanumerics and underscore. In
-     * POSIX mode, the accented characters in the Latin-1 set are also included.
-     */
-    this.wordchars = 'abcdfeghijklmnopqrstuvwxyz' +
-                     'ABCDEFGHIJKLMNOPQRSTUVWXYZ' +
-                     '0123456789' +
-                     '_'
-    if (this.posix) {
-      this.wordchars += 'ßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýþÿ' +
-                        'ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖØÙÚÛÜÝÞ'
-    }
```

Comment:
All of this stuff was for future improvements in mirroring the behavior of the Python `shlex` module. However, since it is not being used, we can go ahead remove it to simplify the module. In the future, if someone comes in asking for some compatibility feature, we'll have to put it back.